Documentation/UserInterface.rst

```diff
@@ -0,0 +1,90 @@
+User Interface
+=============
+
+
+
+For programs that use the "official" application interface (e.g. via the Katydid application) a user sets options in a program using either a configuration file, or command-line options.  Both of these mechanisms allow the user to change the parameters of a program without recompiling any code.
+
+Before getting to the configuration files or command-line options, there are three important built-in command line options that all users should be aware of:
+* **-h [--help]** -- Print the help message for the executable and exit.
+* **-v [--version]** -- Print the version information for the executable and exit.
+* **-c [--config] /path/to/config/file.json** -- Specifies the path to the configuration file that should be used.
+
+## Configuration Files
+Configuration files are broken up into two main sections:
+* Configuring the processor toolbox, including defining which processors will be used, how they are connected, and which processor(s) is in charge of the run,
+* Configuring each individual processor.
+
+Here is an example of an annotated JSON-formatted configuration file::
```

Comment:
Since YAML is the more commonly-used config file format now, can we put in the YAML version of this instead?

Reply:
I'm guessing someone else can do this faster, so I'll go through the rest of the list and come back to this.